etl/helpers.py

```diff
@@ -180,6 +180,10 @@ def __init__(self, __file__: str):
         if len(self.f.relative_to(paths.STEP_DIR).parts) == 6:
             self.f = self.f.parent
 
+        # Start by assuming the current step is public.
+        # Then, if the step is not found in the dag, but it's found as private, is_private will be set to True.
+        self.is_private = False
```

Comment:
I like the idea of using `is_private` as an attribute. How about adding it as an optional argument to `__init__` with default `None` for auto-detection? (you could just call `self.step` to set automatically if `is_private is None`)

Reply:
Thanks! I've just commited a change, please let me know if that's what you meant.